msteams-platform/toolkit/VS-TeamsFx-preview-and-customize-app-manifest.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Teams App Manifest in Teams Toolkit for Visual Studio
+author: surbhigupta
+description: In this module, learn how to edit, preview and customize Teams App Manifest in the different environment for Visual Studio.
+ms.author: nliu
+ms.localizationpriority: medium
+ms.topic: overview
+ms.date: 05/13/2022
+---
+
+# Edit Teams app manifest using Visual Studio
+
+Teams Toolkit in Visual Studio loads manifest from `manifest.template.json` with configurations from `state.{env}.json` and `config.{env}.json` while provisioning and preparing app dependencies. This allows you to create Microsoft Teams app in [Developer Portal](https://dev.teams.microsoft.com/apps) with the manifest.
+
+After scaffolding, in the manifest template file under `templates/appPackage` folder,
+`manifest.template.json` is shared between local and remote environment.
+
+In the manifest template, select **Project** > **Teams Toolkit** > **Open Manifest File**.
+
+:::image type="content" source="../assets/images/Tools-and-SDK-revamp/edit-manifest-for-visual-studio/vs-open-manifest.png" alt-text="Open manifest file":::
+
+### Customize app manifest in Teams Toolkit
+
+There are two types of placeholders in `manifest.template.json`:
+
+- `{{state.xx}}` is pre-defined placeholder, whose value is resolved by Teams Toolkit, defined in `state.{env}.json`. You can't modify the values in `state.{env}.json`.
+- `{{config.manifest.xx}}` is customized placeholder, whose value is resolved from `config.{env}.json`.
+
+You can have a customized parameter by:
+
+- Adding a placeholder in `manifest.template.json` with pattern: `{{config.manifest.xx}}`.
+- Adding a config value in `config.{env}.json`.
+
+    ```
+        {
+            "manifest": {
+            "KEY": "VALUE"
+            }
+    }
+    ```
+
+### Preview app manifest in Teams Toolkit
+
+You can preview values in app manifest in two ways:
+
+- When you hover over the placeholder in `manifest.template.json`, the values for **dev** and **local** environment can be seen.
+
+   :::image type="content" source="../assets/images/Tools-and-SDK-revamp/edit-manifest-for-visual-studio/vs-hover-placeholder1.png" alt-text="Hover over placeholder":::
+
+- You can also hover over the key besides each placeholder in `manifest.template.json`, the same values for **dev** and **local** environment can be seen.
+
+   :::image type="content" source="../assets/images/Tools-and-SDK-revamp/edit-manifest-for-visual-studio/vs-hover-key-placeholder.png" alt-text="Hover over key beside placeholder":::
+
+   > [!NOTE]
+   > If the environment has not been provisioned, or the Teams app dependencies has not been prepared, it indicates that the values for placeholder have not been generated.
```

Comment:
also feel confused about this sentence :(
do you mean that values for placeholders will only be generated after the environment has been provisioned, and the Teams app dependencies has not been prepared?